Modules/Registration/RegistrationMethodsv4/include/itkImageRegistrationMethodv4.hxx

```diff
@@ -102,25 +102,10 @@ ImageRegistrationMethodv4<TFixedImage, TMovingImage, TTransform, TVirtualImage,
   // Default to a 3-level image registration: force the initialization of the ivars that depend on the number of levels.
   this->SetNumberOfLevels(3);
 
-  this->m_ShrinkFactorsPerLevel.resize(this->m_NumberOfLevels);
-  ShrinkFactorsPerDimensionContainerType shrinkFactors;
-  shrinkFactors.Fill(2);
-  this->m_ShrinkFactorsPerLevel[0] = shrinkFactors;
-  shrinkFactors.Fill(1);
-  this->m_ShrinkFactorsPerLevel[1] = shrinkFactors;
-  shrinkFactors.Fill(1);
-  this->m_ShrinkFactorsPerLevel[2] = shrinkFactors;
-
-  this->m_SmoothingSigmasPerLevel.SetSize(this->m_NumberOfLevels);
-  this->m_SmoothingSigmasPerLevel[0] = 2;
-  this->m_SmoothingSigmasPerLevel[1] = 1;
-  this->m_SmoothingSigmasPerLevel[2] = 0;
```

Comment:
Hey @ntustison I would be grateful if you could review this PR as time permits, especially he changes in commit 0500c27391512b1fffc686ba8945aef8018de8ec. The `SetNumberOfLevels` call in https://github.com/InsightSoftwareConsortium/ITK/pull/3876/files#diff-63900e690554f3dd0bd22b3205f700be44c592699ca21a4a48b5f184ab8e61cbL103 initializes the above values to some given values https://github.com/InsightSoftwareConsortium/ITK/pull/3876/files#diff-63900e690554f3dd0bd22b3205f700be44c592699ca21a4a48b5f184ab8e61cbR890, but the lines above override such values (I have not removed the existing body yet, but eventually the `Initialize*` calls should remain only, once the latter as well get polished following your comments). I have not looked into its details, but I am wondering whether that was intended or not, and why, or which would be your recommendations to initialize the values.

Also, I'm wondering whether a rule could be designed for the case where the number of levels is not 3.

Thanks.